### PR TITLE
lib/posix-socket: Change argument type for posix_socket_close

### LINKED ISSUE #646: Socket syscall implementation crashes when there is no file descriptor available
### Describe the bug

When `posix_socket_alloc_fd` fails in `socket.c`, `posix-socket` lib, the cleanup function receives a wrong parameter.  Because of this, when trying to access a member of that parameter, the program crashes.

### Steps to reproduce

_No response_

### Expected behavior

socket should return `-ENFILE`.(23)

### Which architectures were you using or does this bug affect?

_No response_

### Which platforms were you using or does this bug affect?

_No response_

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by flpostolache**
This PR fixes a possible crash in the socket, accept and socketpair syscall implementation when there is no fd available. It changes the structure given as argument to the
posix_socket_close function so that when calling this function it won't result in a crash.

Signed-off-by: Florin Postolache <florin.postolache80@gmail.com>

### Prerequisite checklist

 - [x] Read the [contribution guidelines](https://unikraft.org/docs/contributing/) regarding submitting new changes to the project;
 - [x] Tested your changes against relevant architectures and platforms;
 - [ ] Ran the [`checkpatch.pl`](https://github.com/unikraft/unikraft/blob/staging/support/scripts/checkpatch.pl) on your commit series before opening this PR;
 - [ ] Updated relevant documentation.

### Base target

 - Architecture(s): N/A
 - Platform(s): N/A
 - Application(s): N/A

### CODE REVIEW

File: lib/posix-socket/socket.c [hunk 1]
```diff
@@ -80,10 +81,16 @@ UK_SYSCALL_R_DEFINE(int, socket, int, family, int, type, int, protocol)
 	trace_posix_socket_create_ret(ret);
 	return ret;
 EXIT_ERR_CLOSE:
-	posix_socket_close(sock);
+		dummy = (struct posix_socket_file){
```
Comment (craciunoiuc) [inline lib/posix-socket/socket.c]:
```suggestion
	dummy = (struct posix_socket_file){
```

**Review by marcrittinghaus — CHANGES_REQUESTED** [review]
Hi @maniatro111,

thanks for submitting the bug fix! 😃 It looks good from my perspective. 

However, please remove the changes to the indentation. It seems you have your editor configured to 4 spaces per tab, but Unikraft uses 8 spaces per tab. The indentation is thus right, and your commit breaks it 😏

image

**Review by craciunoiuc — CHANGES_REQUESTED** [review]
Hey

You have an extra Tab on one of the lines.

Also, please specify in the PR and/or the commit that you also fixed some formatting in the file (probably this was done automatically by your editor :sweat_smile: )

**Comment by marcrittinghaus** [comment]
> Hey
> 
> You have an extra Tab on one of the lines.
> 
> Also, please specify in the PR and/or the commit that you also fixed some formatting in the file (probably this was done automatically by your editor 😅 )

Good morning @craciunoiuc,

seems we were both looking at it at the same time 😏 As I said, formatting is generally right. So please revert changes. Maybe @craciunoiuc you have your editor not configured right as well 😆

**Comment by flpostolache** [comment]
Done, @marcrittinghaus and @craciunoiuc. Please have another look. I've disabled the auto indent and reconfigured the editor. Should be ok now.

File: lib/posix-socket/socket.c (hunk 1 above)
Comment (flpostolache) [reply lib/posix-socket/socket.c]: Changed.

**Review by craciunoiuc — APPROVED** [review]
All good here. Thanks!

Reviewed-by: Cezar Craciunoiu <cezar.craciunoiu@unikraft.io>

**Review by marcrittinghaus — APPROVED** [review]
Thanks!

Approved-by: Marc Rittinghaus <marc.rittinghaus@unikraft.io>

**Pull request closed without merging**